its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -1212,6 +1212,26 @@ void checkMultiLanguageSupportWithNonSdkFormat() throws Exception {
         tuple("plsql:S1134", "MultiLanguageSupportNonSdk:MultiLanguageSupportNonSdk/NotIncluded.sql"));
   }
 
+  @Test
+  void checkSourcesTestsIgnored() throws Exception {
+    String projectName = "SourcesTestsIgnored";
+    Path projectDir = TestUtils.projectDir(basePath, projectName);
+    ORCHESTRATOR.getServer().provisionProject(projectName, projectName);
+    String token = TestUtils.getNewToken(ORCHESTRATOR);
+
+    ORCHESTRATOR.executeBuild(TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
+      .addArgument("begin")
+      .setProjectKey(projectName)
+      .setProjectName(projectName)
+      .setProperty("sonar.sources", "Program.cs") // user-defined sources and tests are not passed to the cli.
+      .setProperty("sonar.tests", "Program.cs")); // If they were passed, it results to double-indexing error.
```

Comment:
I tested this locally by deleting the `AnalysisScopeProcessor` and it failed with double indexing :)